plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/HiveMetastore.java

```diff
@@ -214,6 +214,11 @@ default void addDynamicPartitions(String dbName, String tableName, List<String>
         throw new UnsupportedOperationException();
     }
 
+    default void alterTable(String databaseName, String tableName, Table table)
```

Comment:
It allows to modify the Hive Table.

For delta tables without explicit location (aka managed tables) we have to update the location in the SerdeParameters. The renameTable doesn't do that for us.

See `DeltaLakeMetadata`

```
Table.Builder tableBuilder = Table.builder(table)
        .setDatabaseName(newTableName.getSchemaName())
        .setTableName(newTableName.getTableName())
        // Explicitly set the location property because hive metastore won't update it
        .setParameters(deltaTableProperties(session, location, false));

// Explicitly set the path in storage because hive metastore won't update it
// Leave the location as is to trigger rename of the table in hive metastore
tableBuilder.getStorageBuilder().setSerdeParameters(ImmutableMap.of(PATH_PROPERTY, location));

metastore.alterTable(session, handle.getSchemaTableName(), tableBuilder.build());
```

Reply:
Sorry i seem to have misread `renameTable`instead of `replaceTable`. I will look into it.

Reply:
The `alterTable` api change has been removed from `HiveMetastore` interface and `replaceTable` has been used instead.